### PR TITLE
Fix for #10836 (clear hash in Magellan w/ deep linking)

### LINKED ISSUE #10836: Magellan w/ deep linking doesn't clear the hash when there's no longer an active link
#### How to reproduce this bug:

On a page with with Magellan and `data-deep-linking` enabled and a top-margin greater than `data-threshold` (defaults to 50px) such that there is no active link until you scroll past that top-margin:

1. Load the page (without a hash).
2. Scroll down to the first Magellan-tracked element such that the window's location updates to that element's ID (e.g., `#first-heading`).
3. Scroll back to the very top (e.g., by pressing HOME on your keyboard).

#### What should happen:

What should happen after each of the steps to reproduce:

1. The window's location should not have a hash.
2. The window's location should have a hash of `#first-heading`.
3. The window's location should not have a hash.

#### What happened instead:
What actually happens after each of the steps to reproduce:

1. The window's location does not have a hash.
2. The window's location has a hash of `#first-heading`.
3. The window's location has a hash of `#first-heading`.

#### Browser(s) and Device(s) tested on:

Chrome 63.0.3239.84 on macOS Sierra (10.12.6)

#### Foundation Version(s) you are using:

6.4.3

### PR DESCRIPTION
**Pull request by nvenegas**
Fixes #10836.

### CODE REVIEW

**Comment by nicolasv** [comment]
I can't run the tests locally: I run into [#217: phantomjs terminated with signal SIGSEGV](https://github.com/nathanboktae/mocha-phantomjs/issues/217).

File: js/foundation.magellan.js [hunk 1]
```diff
@@ -165,16 +165,21 @@ class Magellan extends Plugin {
...
 
     if(this.options.deepLinking){
       var hash = "";
-      if(curIdx != undefined){
+      if(curIdx !== undefined){
         hash = this.$active[0].getAttribute('href');
       }
       if(hash !== window.location.hash) {
         if(window.history.pushState){
-          window.history.pushState(null, null, hash);
```
Comment (nicolasv) [inline js/foundation.magellan.js]: Previously, when there was no active element (`curIdx === undefined`) we would `window.history.pushState(null, null, '')` which is effectively a no-op.

**Comment by nicolasv** [comment]
> I can't run the tests locally: I run into #217: phantomjs terminated with signal SIGSEGV.

Pointing mocha-phantomjs to my homebrew installed phantomjs binary works makes them run:

```diff
diff --git a/package.json b/package.json
index 5abc54fc2..03efc2726 100644
--- a/package.json
+++ b/package.json
@@ -12,7 +12,7 @@
     "test:ci": "npm run test:sass && npm run test:javascript:ci",
     "test:sass": "mocha test/sass/test_sass.js",
     "test:javascript:transpile": "gulp sass:foundation && gulp test:transpile-js",
-    "test:javascript:phantomjs": "npm run test:javascript:transpile && mocha-phantomjs --ignore-resource-errors test/javascript/index.html",
+    "test:javascript:phantomjs": "npm run test:javascript:transpile && mocha-phantomjs -p /usr/local/bin/phantomjs --ignore-resource-errors test/javascript/index.html",
     "test:javascript:browserstack": "npm run test:javascript:transpile && browserstack-runner",
     "test:javascript:ci": "npm run test:javascript:transpile && mocha-phantomjs --ignore-resource-errors test/javascript/index.html && browserstack-runner",
     "test:visual": "gulp test",
```

**Comment by nicolasv** [comment]
Manually testing the `magellan/active-item.html` visual test (via `npm run test:visual`) looks good: scrolling back to the _Pre-anchor Section_ clears the location hash.

File: js/foundation.magellan.js [hunk 2]
```diff
@@ -165,21 +165,26 @@ class Magellan extends Plugin {
...
         hash = this.$active[0].getAttribute('href');
       }
       if(hash !== window.location.hash) {
-        if (window.history.pushState) {
-          if (this.options.updateHistory) {
-            window.history.pushState({}, '', hash);
-          } else {
-            window.history.replaceState({}, '', hash);
+        if(window.history.pushState){
+          var url = curIdx !== undefined ? hash : window.location.pathname;
```
Comment (ncoden) [inline js/foundation.magellan.js]: `window.location.pathname` is only `/path`, but an url can contains: `/path?query#fragment`. So the query params will be lost

**Review by ncoden — COMMENTED** [review]
Found a bug, but everything else looks good. 👍

**Review by ncoden — APPROVED** [review]
Fixed 👍

**Comment by ncoden** [comment]
Merged. Thank you @nicolasv for you first PR 🎉

**Comment by nicolasv** [comment]
Thanks, @ncoden!

**Pull request merged**